### ISSUE TITLE
a11y: Headings inconsistent & out of semantic order on results page

### ISSUE THREAD

**Issue by iamjolly** [18F/all_sorns#139]
## Description:
Headings on the page are not nested consistently to provide a semantic match with the page's information structure conveyed visually. 
- The first heading level 2 on the page should be the search input's `label` instead of an `h2` element. 
- The `h3` for "Refine your search" has `h2` elements for "Sections" and "Agencies" which are subordinate to the `h3`, so they should be `h4` or not headings at all (see video below about how screen readers encounter those).
- The results cards in the `main` content on the page has an `h2` for the result title (good), but then skips `h3` and uses `h4` for the "FOUND IN" content header. That should be an `h3` to reduce potential confusion for some users.

## Screenshots:
![screenshot of sorns results headings](https://user-images.githubusercontent.com/1093423/105394414-c7527480-5bda-11eb-9d28-1421a4cf274c.png)
![screenshot of sorns refine results buttons](https://user-images.githubusercontent.com/1093423/105395069-932b8380-5bdb-11eb-9099-9283366c7d04.png)

## Screencast/video:
https://user-images.githubusercontent.com/1093423/105395335-d84fb580-5bdb-11eb-90be-6315ca02b3f9.mp4

## WCAG SC
[Success Criterion 1.3.1 Info and Relationships (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships): Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.

## Recommended fix(es):
1. Removing the `h2` and using `label` for the search input field per #129 will resolve the first `h2` issue.
2. Change the `h3` for **Refine your search** to an `h2`
3. Replace the `h2` with `div` elements for **Sections** and **Agencies**
